pypose/module/dynamics.py

```diff
@@ -0,0 +1,342 @@
+        input (:obj:`Tensor`): The input of the current timestamp of LTI system.
+
+    Return:
+        Tuple of Tensors: The state of the next timestamp (state-transition) and the system output (observation).
+
+    Every linear time-invariant lumped system can be described by a set of equations of the form 
+    which is called the state-space equation.
+
+    .. math::
+        \begin{align*}
+            \mathbf{z} = \mathbf{A}\mathbf{x} + \mathbf{B}\mathbf{u} + \mathbf{c}_1 \\
+            \mathbf{y} = \mathbf{C}\mathbf{x} + \mathbf{D}\mathbf{u} + \mathbf{c}_2 \\
+        \end{align*}
+
+    where we use :math:`\mathbf{x}` and :math:`\mathbf{u}` to represent state and input of the current timestamp of LTI system.
+            
+    Here, we consider the discrete-time system dynamics.  
+        
+    Note:
+        According to the actual physical meaning, the dimensions of A, B, C, D must be the consistent,
+        whether in batch or not.
+
+        :math:`\mathbf{A}`, :math:`\mathbf{B}`, :math:`\mathbf{C}`, :math:`\mathbf{D}`, :math:`\mathbf{x}`, :math:`\mathbf{u}` 
+        could be a single input or in a batch. In the batch case, their dimensions must be consistent 
+        so that they can be multiplied for each channel.
+             
+        Note that here variables are given as row vectors.
+    '''
+    
+    def __init__(self, A, B, C, D, c1=None, c2=None):
+        super(LTI, self).__init__(time=False)
+        assert A.ndim == B.ndim == C.ndim == D.ndim, "Invalid System Matrices dimensions"
+        self.A, self.B, self.C, self.D = A, B, C, D
+        self.c1, self.c2 = c1, c2
+    
+    def forward(self, state, input):
+        r'''
+
+        Example:
+            >>> A = torch.randn((2,3,3))
+                B = torch.randn((2,3,2))
+                C = torch.randn((2,3,3))
+                D = torch.randn((2,3,2))
+                c1 = torch.randn((2,1,3))
+                c2 = torch.randn((2,1,3))
+                state = torch.randn((2,1,3))
+                input = torch.randn((2,1,2))
+            >>> lti = pp.module.LTI(A, B, C, D, c1, c2)
+            >>> lti(state,input)
+            tensor([[[-8.5639,  0.0523, -0.2576]],
+                    [[ 4.1013, -1.5452, -0.0233]]]), 
+            tensor([[[-3.5780, -2.2970, -2.9314]],
+                    [[-0.4358,  1.7306,  2.7514]]]))
+
+        Note:
+            In this general example, all variables are in a batch. User definable as appropriate.
+        '''
+        if self.A.ndim >= 3:
+            assert self.A.ndim == state.ndim == input.ndim,  "Invalid System Matrices dimensions"
+        else:
+            assert self.A.ndim == 2,  "Invalid System Matrices dimensions"
+
+        z = self.state_transition(state, input)
+        y = self.observation(state, input)
+        return z, y
+
+    def state_transition(self, state, input):
+        return state.matmul(self.A.mT) + input.matmul(self.B.mT) + self.c1
+
+    def observation(self, state, input):
```

Comment:
add a short doc:
```
Perform one step for linear system observation.
```
maybe we can also move the linear observation function here.